src/183/test/kotlin/org/rust/ide/runAnything/CargoRunAnythingProviderTest.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.runAnything
+
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.actionSystem.impl.SimpleDataContext
+import org.rust.EmptyDescriptor
+import org.rust.ProjectDescriptor
+import org.rust.RsTestBase
+
+class CargoRunAnythingProviderTest : RsTestBase() {
+    private lateinit var provider: CargoRunAnythingProvider
+    private lateinit var dataContext: DataContext
+
+    override fun setUp() {
+        super.setUp()
+        provider = CargoRunAnythingProvider()
+        dataContext = SimpleDataContext.getProjectContext(project)
+    }
+
+    @ProjectDescriptor(EmptyDescriptor::class)
+    fun `test do not provide values no rust project linked`() {
+        assertEmpty(provider.getValues(dataContext, "cargo"))
+    }
+
+    fun `test do not provide values for non-cargo command patterns`() {
+        assertEmpty(provider.getValues(dataContext, ""))
+    }
+
+    fun `test command completion`() {
+        val commands = listOf(
+            "build", "check", "clean", "doc", "run", "test", "bench", "update", "search", "publish", "install"
+        ).map { "cargo $it" }
+        assertSameElements(provider.getValues(dataContext, "cargo"), commands)
+        assertSameElements(provider.getValues(dataContext, "cargo "), commands)
+        assertSameElements(provider.getValues(dataContext, "cargo ru"), emptyList())
```

Comment:
Are you sure that it's desired behavior?
Personally, I expected at least `cargo run` in completion list for such input or even several `cargo run` with some args